lib/vmdb/settings.rb

```diff
@@ -88,13 +88,26 @@ def self.build_template
     private_class_method :build_template
 
     def self.build(resource)
-      build_template.tap do |settings|
-        settings.add_source!(DatabaseSource.new(resource))
+      objects_in_settings_chain = resource ? resource.objects_in_settings_chain : []
+      build_without_local(objects_in_settings_chain).tap do |settings|
         local_sources.each { |s| settings.add_source!(s) } if resource.try(:is_local?)
       end
     end
     private_class_method :build
 
+    def self.parent_settings(resource)
+      objects_in_settings_chain = resource.objects_in_settings_chain
+      build_without_local(objects_in_settings_chain[0...-1]).load!
+    end
+    private_class_method :parent_settings
+
+    def self.build_without_local(objects_in_settings_chain)
+      build_template.tap do |settings|
+        objects_in_settings_chain.each { |r| settings.add_source!(DatabaseSource.new(r)) }
```

Comment:
As discussed, I think there might be a problem here because the Zone object will get cached.  Then, if the server's zone is changed, then it won't pick up the new zone.

Instead, I think we discussed only caching the miq_server object, and then having a way for each DatabaseSource to get what it needs from the miq_server object...

For example, for zone, the DatabaseSource#load can do `@resource.zone.reload.settings_chages.each ...`, and it will be guaranteed to get the right zone.